File: DoAnTotNghiep/Kymdan/src/main/java/com/kymdan/backend/services/customer/CustomerService.java
package com.kymdan.backend.services.customer;

import com.kymdan.backend.entity.Customer;
import com.kymdan.backend.model.AppUserDTO;
import com.kymdan.backend.model.MessageDTO;

public interface CustomerService {
    Customer findByEmail(String email);

    Customer findByName(String name);

    MessageDTO editInformation(AppUserDTO appUserDTO);

    MessageDTO editPassword(String username, String oldPassword, String newPassword);
}

File: DoAnTotNghiep/Kymdan_front_end/src/app/service/customer.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class CustomerService {
  readonly API = 'http://localhost:8080/customer';

  constructor(public http: HttpClient) {
  }

  getInformationCustomer(name: any): Observable<any> {
    return this.http.get(this.API + '/information/' + name);
  }

  editInformation(newInformation): Observable<any> {
    return this.http.post(this.API + '/edit-information', newInformation, {headers: {skip: 'true'}});
  }

  register(account): Observable<any> {
    return this.http.post(this.API + '/register', account, {headers: {skip: 'true'}});
  }

  editPassword(username, oldPassword, newPassword): Observable<any> {
    return this.http.get(this.API + '/edit-password/' + username + '/' + oldPassword + '/' + newPassword);
  }
}
